src/database.c

```diff
@@ -184,6 +187,29 @@ static void abort_transaction(MDB_txn *txn)
 	dbi_init = 0;
 }
 
+/*
+ * Convert path to a hash value. Used when the path exceeds the LMDB key limit(511).
+ * Note: Returned value must be deallocated.
+ */
+char *path_to_hash(const char *path)
+{
+    gcry_md_hd_t h;
+    unsigned int len;
+    char *digest, *hptr;
+
+    gcry_md_open(&h, GCRY_MD_SHA256, GCRY_MD_FLAG_SECURE); /* initialize the hash context */
+    gcry_md_write(h, path, strlen(path)); /* hash the text */
```

Comment:
Generally, I'd only provide comments for things that are not immediately obvious. If you're wanting to outline function flow, you probably want to do that in the comment above the function. This is just my recommendation.